module/VuFind/src/VuFind/Db/Service/UserListService.php

```diff
@@ -142,6 +169,33 @@ public function getUserListsAndCountsByUser(UserEntityInterface|int $userOrId):
         return $result;
     }
 
+    /**
+     * Get lists associated with a particular tag and/or list of IDs. If IDs and
+     * tags are both provided, only the intersection of matches will be returned.
+     *
+     * @param string|string[]|null $tag        Tag or tags to match (by text, not ID; null for all)
+     * @param int|int[]|null       $listId     List ID or IDs to match (null for all)
+     * @param bool                 $publicOnly Whether to return only public lists
+     * @param bool                 $andTags    Use AND operator when filtering by tag.
+     *
+     * @return UserListEntityInterface[]
+     */
+    public function getUserListsByTagAndId(
+        string|array|null $tag = null,
+        int|array|null $listId = null,
+        bool $publicOnly = true,
+        bool $andTags = true
+    ): array {
+        $listIds = array_column(
+            iterator_to_array($this->getDbTable('ResourceTags')->getListsForTag($tag, $listId, $publicOnly, $andTags)),
```

Comment:
`getUserListsByTagAndId()` allows `$tag` to be `null`, but `getListsForTag()` does not?

Reply:
Good catch; I think it was working in spite of this, but I've added the support more explicitly in commit 0d723bfd870c0b7d780eb3976a1b591f21aa0fbd.